ui/app/components/secure-variable-form.js

```diff
@@ -75,10 +111,16 @@ export default class SecureVariableFormComponent extends Component {
 
   @action
   async save(e) {
-    e.preventDefault();
+    if (e.type === 'submit') {
+      e.preventDefault();
+    }
+    // TODO: temp, hacky way to force translation to tabular keyValues
+    if (this.view === 'json') {
+      this.translateAndValidateItems('table');
+    }
     try {
-      const nonEmptyItems = this.keyValues.filter(
-        (item) => item.key.trim() && item.value
+      const nonEmptyItems = A(
+        this.keyValues.filter((item) => item.key.trim() && item.value)
```

Comment:
`thought` `non-blocking`:  the `filter` method and associated length check in the conditional aren't asynchronous callbacks and technically don't belong in the `try/catch` block. I can totally understand that we're throwing an error that should be swallowed by the `catch` and then show a flash method.

But, since that check is synchronous in nature and we're writing a form, shouldn't the submit button just be disabled with validation text? IMO, flash messages are here to help us display information after an asynchronous call is made and communicate status of the async request.

Reply:
I think that is a good pattern so long as this was just a form; but since we're also including JSON editing capabilities, I worry about having a submit button that flickers in/out of disabled state as a user types. Let me think about this one a bit more; if I can make it appear reasonably to the end user, I'll make this a `formHasNoNonEmptyItems` property or something similar.

Reply:
**Chants uncontrollably** "Statecharts, Statecharts, Statecharts!"